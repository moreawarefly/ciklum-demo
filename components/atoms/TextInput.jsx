import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Label = styled.label`
  font-weight: bold;
  font-size: .9rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Input = styled.input`
  margin-left: .5rem;
  background-color: #eee;
  border: none;
  border-radius: 3px;
  padding: .5rem;
  font-size: 1rem;
  box-sizing: border-box;
  width: 150px;
`;

function TextInput({
  className,
  required,
  type,
  label,
  onChangeHandler,
}) {
  return (
    <Label className={className}>
      {!!label && `${label}: `}
      <Input
        required={required}
        type={type}
        onChange={onChangeHandler}
      />
    </Label>
  );
}

TextInput.defaultProps = {
  className: '',
  required: false,
  type: 'text',
  label: null,
  onChangeHandler: () => {},
};

TextInput.propTypes = {
  className: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  label: PropTypes.string,
  onChangeHandler: PropTypes.func,
};

export default TextInput;
