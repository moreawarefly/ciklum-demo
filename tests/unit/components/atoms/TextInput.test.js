import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../../../../components/atoms/TextInput';

describe('Dropdown', () => {
  it('renders correctly w/o arguments', () => {
    const tree = renderer
      .create(<TextInput />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with provided all attributes', () => {
    const tree = renderer
      .create(
        <TextInput
          name="mockName"
          required
          type="number"
          label="mockLabel"
          onChangeHandler={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
