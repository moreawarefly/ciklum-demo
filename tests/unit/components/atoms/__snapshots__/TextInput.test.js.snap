// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly w/o arguments 1`] = `
<label
  className="TextInput__Label-wh9sup-0 bszEGB"
>
  <div
    className="TextInput__InputWrapper-wh9sup-1 kMtFrB"
    required={false}
  >
    <input
      className="TextInput__Input-wh9sup-2 fYtAcX"
      name=""
      onChange={[Function]}
      required={false}
      type="text"
    />
  </div>
</label>
`;

exports[`Dropdown renders correctly with provided all attributes 1`] = `
<label
  className="TextInput__Label-wh9sup-0 bszEGB"
>
  mockLabel: 
  <div
    className="TextInput__InputWrapper-wh9sup-1 cDbVNq"
    required={true}
  >
    <input
      className="TextInput__Input-wh9sup-2 fYtAcX"
      name="mockName"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
</label>
`;
